### DISCUSSION TITLE
Installation warning

### DISCUSSION THREAD

**Discussion by David-Maran** [timschofield/webERP#203] [Installation Issues]
Warning when install:
`Deprecated: Optional parameter $NewSQL declared before required parameter $DBType is implicitly treated as a required parameter in C:...\webERP-master\install\index.php on line 1092`
the function : 
`function PopulateSQLData($NewSQL=false,$Demo=false,$db,$DBType,$NewDB = false)`
will be : 
`function PopulateSQLData($NewSQL = false, $Demo=false, $db = null, $DBType = null, $NewDB = false)`
or : 
`function PopulateSQLData($db, $DBType, $NewSQL = false, $Demo = false, $NewDB = false)`

**Comment by timschofield** [comment]
Hi, is this from the latest release (4.15.2) or from the latest code from the master branch on github?

Thanks
Tim

**Comment by David-Maran** [comment]
From the master branch

**Comment by timschofield** [comment]
I have rewritten the installer that is in the master branch now. Would be great if you could test it.

Thanks
Tim

**Comment by David-Maran** [comment]
Sorry but I will not be using this app, it has too many things wrong with it, so you will not get any feedback for your tests.

**Comment by timschofield** [comment]
OK, no problem.
Seems curious that you won't test whether the things that were wrong have benn fixed, because of the things that are wrong, but that is your prerogative.

Thanks for your interest anyway
Tim

**Comment by David-Maran** [comment]
I tested your new installer, yes the user interface changes but some failures reappear and the installation crashes due to bad charset collation. I'm not here to rebuild  this app who came with same problems as frontaccounting.

**Comment by timschofield** [comment]
I am surprised at that David. There were initially a couple of table
definitions with bad charset collation, but I corrected them last night.
Can you tell me the tables that failed?

On Sat, 26 Oct 2024 at 13:46, David Baron ***@***.***> wrote:

> I tested your new installer, yes the user interface changes but some
> failures reappear and the installation crashes due to bad charset
> collation. I'm not here to rebuild this app who came with same problems as
> frontaccounting.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/timschofield/webERP/discussions/203#discussioncomment-11060505>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAL6LBY57O4IMEU4WDNRLDZ5OMKJAVCNFSM6AAAAABQUTENIGVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCMBWGA2TANI>
> .
> You are receiving this because you modified the open/close state.Message
> ID: ***@***.***
> com>
>


-- 
www.weberpafrica.com
@TimSchofield2
Blog: https://kwamoja.home.blog/

**Comment by David-Maran** [comment]
Ah yes you modified them, it was 'jnltmpldetails' and 'jnltmplheader'.

Ok let me test again the installer.

**Comment by timschofield** [comment]
Thanks David

On Sat, 26 Oct 2024 at 15:14, David Baron ***@***.***> wrote:

> Ah yes you modified them, it was 'jnltmpldetails' and 'jnltmplheader'.
>
> Ok let me test again the installer.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/timschofield/webERP/discussions/203#discussioncomment-11060922>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAL6LDSMJ6HDERG4JCKC7TZ5OWWFAVCNFSM6AAAAABQUTENIGVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCMBWGA4TEMQ>
> .
> You are receiving this because you modified the open/close state.Message
> ID: ***@***.***
> com>
>


-- 
www.weberpafrica.com
@TimSchofield2
Blog: https://kwamoja.home.blog/

**Comment by David-Maran** [comment]
Install ok (I think). But:
impossible to login
if I enable error reporting:
infinite loop and error: GET /Z_UpgradeDatabase.php - Uncaught TypeError: count(): Argument #1 ($value) must be of type Countable|array , null given in webERP-master\includes\header.php:123

**Comment by timschofield** [comment]
Curious. I've just done 2 clean installs one with and one without the demo
data, and they both login fine after install.

How many tables were installed?

On Sat, 26 Oct 2024 at 15:26, David Baron ***@***.***> wrote:

> Install ok (I think). But:
> impossible to login
> if I enable error reporting:
> infinite loop and error: GET /Z_UpgradeDatabase.php - Uncaught TypeError:
> count(): Argument #1 <https://github.com/timschofield/webERP/pull/1>
> ($value) must be of type Countable|array , null given in
> webERP-master\includes\header.php:123
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/timschofield/webERP/discussions/203#discussioncomment-11060978>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAL6LGZAEZ36S6VZELELQDZ5OYC3AVCNFSM6AAAAABQUTENIGVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCMBWGA4TOOA>
> .
> You are receiving this because you modified the open/close state.Message
> ID: ***@***.***
> com>
>


-- 
www.weberpafrica.com
@TimSchofield2
Blog: https://kwamoja.home.blog/

**Comment by David-Maran** [comment]
178 tables lol
I think something strange happened ...

**Comment by timschofield** [comment]
Someone else had the same trouble -
https://github.com/timschofield/webERP/discussions/204

I have just suggested he change the value in the config table where
confname='DBUpdateNumber' value should be 7 not 8

On Sat, 26 Oct 2024 at 17:47, David Baron ***@***.***> wrote:

> 178 tables lol
> I think something strange happened ...
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/timschofield/webERP/discussions/203#discussioncomment-11061807>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAL6LBLGU7C6Y3WFQYNXXLZ5PITZAVCNFSM6AAAAABQUTENIGVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCMBWGE4DANY>
> .
> You are receiving this because you modified the open/close state.Message
> ID: ***@***.***
> com>
>


-- 
www.weberpafrica.com
@TimSchofield2
Blog: https://kwamoja.home.blog/

**Comment by dalers** [comment]
closing as no longer relevent
